valinta-tulos-service/src/main/scala/fi/vm/sade/valintatulosservice/migri/MigriService.scala

```diff
@@ -0,0 +1,99 @@
+import fi.vm.sade.valintatulosservice.tarjonta.{HakuService, HakukohdeMigri}
+import fi.vm.sade.valintatulosservice.valintarekisteri.db.ValinnantulosRepository
+import fi.vm.sade.valintatulosservice.valintarekisteri.domain.{HakijaOid, HakukohdeOid, HyvaksyttyValinnanTila, ValinnantulosWithTilahistoria}
+import fi.vm.sade.valintatulosservice.{AuditInfo, LukuvuosimaksuService, ValinnantulosService}
+
+import scala.collection.immutable.Set
+import scala.collection.mutable
+
+class MigriService(hakemusRepository: HakemusRepository, hakuService: HakuService, valinnantulosService: ValinnantulosService, oppijanumerorekisteriService: OppijanumerorekisteriService, valintarekisteriService: ValinnantulosRepository, lukuvuosimaksuService: LukuvuosimaksuService) extends Logging {
+
+  private def getHakijat(hakijaOids: Set[HakijaOid]): Set[MigriHakija] = {
+    oppijanumerorekisteriService.henkilot(hakijaOids).fold(_ => throw new IllegalArgumentException(s"No migri hakijas found for oid: $hakijaOids found."), henkilot => {
+      henkilot.map(henkilo =>
+        MigriHakija(
+          henkilotunnus = if (henkilo._2.hetu.nonEmpty) henkilo._2.hetu.toString else null,
+          henkiloOid = henkilo._2.oid.toString,
+          sukunimi = if (henkilo._2.sukunimi.nonEmpty) henkilo._2.sukunimi.toString else null,
+          etunimet = if (henkilo._2.etunimet.nonEmpty) henkilo._2.etunimet.toString else null,
+          kansalaisuudet = if (henkilo._2.kansalaisuudet.nonEmpty) henkilo._2.kansalaisuudet.get else null,
+          syntymaaika = if (henkilo._2.syntymaaika.nonEmpty) henkilo._2.syntymaaika.toString else null,
+          mutable.Set()
+        )
+      ).filterNot(hakija => hakija.kansalaisuudet.contains("246")).toSet
+    })
+  }
+
+  def getHakemuksetByHakijaOids(hakijaOids: Set[HakijaOid], auditInfo: AuditInfo): Set[MigriHakija] = {
+    val foreignHakijat: Set[MigriHakija] = getHakijat(hakijaOids)
+    foreignHakijat.foreach(hakija => {
+      val hyvaksytyt: Set[HyvaksyttyValinnanTila] = valintarekisteriService.getHakijanHyvaksytValinnantilat(HakijaOid(hakija.henkiloOid))
+      val hyvaksytytHakemusOidit = hyvaksytyt.map(h => h.hakemusOid)
+      val hyvaksytytHakukohdeOidit = hyvaksytyt.map(h => h.hakukohdeOid)
+      valinnantulosService.getValinnantuloksetForHakemukset(hyvaksytytHakemusOidit, auditInfo) match {
+        case tulokset: Set[ValinnantulosWithTilahistoria] =>
+          tulokset
+            .filter(t => t.valinnantulos.isHyvaksytty)
+            .map { tulos =>
+              getHakukohdeMigri(tulos.valinnantulos.hakukohdeOid) match {
+                case Some(hakukohde: HakukohdeMigri) =>
+                  if (hyvaksytytHakukohdeOidit.contains(hakukohde.oid)) {
+                    hakemusRepository.findHakemus(tulos.valinnantulos.hakemusOid).fold(e =>
+                      logger.warn(s"No hakemus found for migri hakijaOid: ${tulos.valinnantulos.hakemusOid}, cause: ${e.toString}"), h => {
```

Comment:
Voiko näin oikeesti käydä usein, ja pitäiskö silloin kaatua jotenkin äänekkäämmin?

Reply:
heitetään poikkeus ja logitellaan error-tasolla.